### PR TITLE
Tests: Stubbable reload util

### PR DESCRIPTION
**Pull request by islemaster**
Some of our tests use sinon to stub `window.location.reload` which works fine in PhantomJS, but fails in Chrome which doesn't allow replacing properties on `window.location`.  This introduces a `utils.reload` helper that does the same thing, but can be easily stubbed in tests.

I've updated the two existing tests that were stubbing `window.location.reload`.  I also intend to use this in one of my open PRs.

### CODE REVIEW

File: apps/test/unit/lib/kits/maker/ui/SetupChecklistTest.js [hunk 1]
```diff
@@ -48,7 +44,8 @@ describe('SetupChecklist', () => {
   });
 
   describe('test with expected console.error', () => {
-    allowConsoleErrors();
+    // Completely squelch console.error calls - ignore them and print nothing
+    beforeEach(() => console.error.reset());
```
Comment (davidsbailey) [inline apps/test/unit/lib/kits/maker/ui/SetupChecklistTest.js]: revert these lines? they seem unrelated to the rest of the PR

Comment (islemaster) [reply apps/test/unit/lib/kits/maker/ui/SetupChecklistTest.js]: :stuck_out_tongue: Fine.  It's intentional cleanup of logspam in this test, but I can break it out.

Comment (davidsbailey) [reply apps/test/unit/lib/kits/maker/ui/SetupChecklistTest.js]: no no no, I only meant to revert this if the change was unintentional. it's fine to keep it in, I just thought it was something you didn't mean to commit. if you've already removed it, that's fine too.
